mods/pageviews.js

```diff
@@ -251,6 +251,12 @@ var validateYearMonthDay = function(rp) {
 PJVS.prototype.pageviewsForArticleFlat = function(restbase, req) {
     var rp = req.params;
 
+    // dates are passed in as YYYYMMDD but we need the HH to match the loaded data
+    // which was originally planned at hourly resolution
+    if (rp.start && rp.end) {
```

Comment:
That's quite fair.  Fixing now.

On Sun, Nov 15, 2015 at 12:35 PM, Marko Obrovac <notifications@github.com>
wrote:

> In mods/pageviews.js
> <https://github.com/wikimedia/restbase/pull/414#discussion_r44874928>:
>
> > @@ -251,6 +251,12 @@ var validateYearMonthDay = function(rp) {
> >  PJVS.prototype.pageviewsForArticleFlat = function(restbase, req) {
> >      var rp = req.params;
> >
> > +    // dates are passed in as YYYYMMDD but we need the HH to match the loaded data
> > +    // which was originally planned at hourly resolution
> > +    if (rp.start && rp.end) {
>
> Instead of working around the fact that you dropped HH from the
> parameters, you should fix the validation logic, since it will still report
> that HH is needed, which will be pretty confusing for users.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wikimedia/restbase/pull/414/files#r44874928>.
>